pkgs/by-name/st/stardust-xr-kiara/package.nix

```
@@ -0,0 +1,54 @@
+{
+  lib,
+  fetchFromGitHub,
+  fetchpatch,
+  writeScript,
+  unstableGitUpdater,
+  rustPlatform,
+  cmake,
+  makeWrapper,
+  pkg-config,
+  niri,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "stardust-xr-kiara";
+  version = "0-unstable-2024-07-07";
+
+  src = fetchFromGitHub {
+    owner = "stardustxr";
+    repo = "kiara";
+    rev = "7daaa0a2e3822d949e6c4abf93af159eae9a544a";
+    hash = "sha256-5j83e2kcCStPgbwAkr3OFjOpJIErXAPJ6z06BlmtuHE=";
+  };
+
+  NIRI_CONFIG = "${src}/src/niri_config.kdl";
+  STARDUST_RES_PREFIXES = "${src}/res";
```

Nitpick: Let's use `__structuredAttrs`-friendly values.

```suggestion
  env = {
    NIRI_CONFIG = "${src}/src/niri_config.kdl";
    STARDUST_RES_PREFIXES = "${src}/res";
  };
```